scene/2d/animated_sprite_2d.cpp

```diff
@@ -361,6 +365,10 @@ bool AnimatedSprite2D::is_flipped_v() const {
 
 void AnimatedSprite2D::_res_changed() {
 	set_frame(frame);
+	Vector<String> animation_names = frames->get_animation_names();
+	if (!animation_names.has(animation) && animation_names.size() != 0) {
+		set_animation(animation_names[0]);
+	}
```

Comment:
`set_frame(frame)` validates the `frame` according to the `animation` so I think it makes more sense to firstly try to make the `animation` valid (if it's possible).
```suggestion
	Vector<String> animation_names = frames->get_animation_names();
	if (!animation_names.has(animation) && animation_names.size() != 0) {
		set_animation(animation_names[0]);
	}
	set_frame(frame);
```